### ISSUE TITLE
Interleaver Initialization

### ISSUE THREAD

**Issue by fstop22** [aff3ct/aff3ct#132]
Hi,

I am trying to build DVB-RCS2 encoder using the bootstrap method described in `my_project_with_aff3ct`.  However, I am running into issue with the initialization of the interleaver.  Below is how I am constructing the encoder in the `init_modules` function:
```
    ///////////////////////////////////////////////////////////////////////////
    // Build CTC encoder.
    // Constituent encoder 1 - natural order
    auto enco_n = new module::Encoder_RSC_DB<B_32>(p.K,
                                                   2 * p.K,
                                                   "DVB-RCS2");

    // Constituent encoder 2 - interleaved order
    auto enco_i = new module::Encoder_RSC_DB<B_32>(p.K,
                                                   2 * p.K,
                                                   "DVB-RCS2");

    // Make the interleaver core then the interleaver
    aff3ct::tools::Interleaver_core_ARP_DVB_RCS2<> intl_core(p.K/2);
    auto intl = new module::Interleaver<B_32>(intl_core);

    // Finally, build the actual encoder
    m.encoder = std::unique_ptr<module::Encoder_turbo_DB<>>(new module::Encoder_turbo_DB<>(p.K,
                                                                                           p.N,
                                                                                           *enco_n,
                                                                                           *enco_i,
                                                                                           *intl));
    ///////////////////////////////////////////////////////////////////////////

```

When the `encode` method of the encoder is called in main processing loop, the following exception is thrown:
```
# ---------------------||------------------------------------------------------||---------------------
#  Signal Noise Ratio  ||   Bit Error Rate (BER) and Frame Error Rate (FER)    ||  Global throughput  
#         (SNR)        ||                                                      ||  and elapsed time   
# ---------------------||------------------------------------------------------||---------------------
# ----------|----------||----------|----------|----------|----------|----------||----------|----------
#     Es/N0 |    Eb/N0 ||      FRA |       BE |       FE |      BER |      FER ||  SIM_THR |    ET/RT 
#      (dB) |     (dB) ||          |          |          |          |          ||   (Mb/s) | (hhmmss) 
# ----------|----------||----------|----------|----------|----------|----------||----------|----------
terminate called after throwing an instance of 'aff3ct::tools::runtime_error'
  what():  In the '/home/jgrasty/git/my_project_with_aff3ct/lib/aff3ct/include/Tools/Interleaver/Interleaver_core.hxx' file at line 53 ('get_lut_inv' function): "The 'init()' method has to be called before trying to get the inverse lookup table."
Backtrace:
./bin/my_project: void aff3ct::module::Encoder<int>::encode<std::allocator<int> >(std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> >&, int, bool) +0x87 [0x45c9b7]
./bin/my_project: main() +0x4d6 [0x457ea2]
/lib64/libc.so.6: __libc_start_main() +0xf3 [0x7ff8163c17b3]
./bin/my_project: _start() +0x2e [0x4578fe]
Aborted (core dumped)
```

I have tried calling `init` method on the interleaver_core object as described in #43; however, the `init` is now `protected`.  Looking at the source code for `Interleaver_core_ARP_DVB_RCS2`, `init` is called in the constructor.

Am I not using or instantiating the interleaver correctly?

Thanks for your help,
Jeff

**Comment by fstop22** [comment]
By the way, I working with the `development` branch.

**Comment by fstop22** [comment]
So, it looks like the problem is that `Interleaver_core_ARP_DVB_RCS2` object goes out of scope when `init_modules` finishes.  But  the signature of `Interleaver` object takes a reference to `Interleaver_core_ARP_DVB_RCS2', but it does not make a copy.
```
explicit Interleaver(const tools::Interleaver_core<T> &core);
```
 So the `Interleaver_core`  no longer exists when the encoder starts to encode and interleave input data.

I made the `Interleaver_core` object global and the sim runs.

However, how can I create an `Interlever_core_ARP_DVB_RCS2` object that will persist?  I tried using `new`, but `Interleaver` construct wants a reference to the `Interleaver_core` object, not a pointer.

Thanks,
Jeff
